Wearable/src/main/java/com/pimpimmobile/librealarm/WearActivity.java

```diff
@@ -118,40 +140,93 @@ public void onCreate(Bundle b) {
                         WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD |
                         WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED |
                         WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON,
-                        WindowManager.LayoutParams.FLAG_FULLSCREEN |
+                WindowManager.LayoutParams.FLAG_FULLSCREEN |
                         WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD |
                         WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED |
                         WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON);
         setContentView(R.layout.wear_activity);
-        mGoogleApiClient = new GoogleApiClient.Builder(this)
-                .addApi(Wearable.API)
-                .addConnectionCallbacks(this)
-                .addOnConnectionFailedListener(this)
-                .build();
-
-        // If attempt fails for some reason, retry in 20 seconds.
-        if (PreferencesUtil.getIsStarted(this)) AlarmReceiver.post(this, 20000);
-
-        mGoogleApiClient.connect();
-        NfcManager nfcManager =
-                (NfcManager) WearActivity.this.getSystemService(Context.NFC_SERVICE);
-        mNfcAdapter = nfcManager.getDefaultAdapter();
-        mNfcAdapter.enableReaderMode(WearActivity.this, new NfcAdapter.ReaderCallback() {
-            @Override
-            public void onTagDiscovered(Tag tag) {
-                new NfcVReaderTask().execute(tag);
-                mNfcAdapter.disableReaderMode(WearActivity.this);
-            }
-        }, NfcAdapter.FLAG_READER_NFC_V, null);
 
-        PowerManager manager = (PowerManager) getSystemService(POWER_SERVICE);
-        mWakeLock = manager.newWakeLock(PowerManager.FULL_WAKE_LOCK |
-                PowerManager.SCREEN_BRIGHT_WAKE_LOCK | PowerManager.ACQUIRE_CAUSES_WAKEUP, "tag");
+        createScripts();
+
+        final long time_now = System.currentTimeMillis();
+        if (time_now > busy) {
+            busy =  time_now+120000; // maximum time we can lock for
+            PowerManager manager = (PowerManager) getSystemService(POWER_SERVICE);
+
+            // TODO wakelock management needs reviewing - is there possiblity to arrive here with this wakelock already held?
```

Comment:
Only if we have two instances of the Activity (which happens sometimes =(   ), though I don't think it's an issue to just release and recreate the wakelock